packages/babel-plugin-remove-undefined-if-possible/src/index.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+function removeRvalIfUndefined(declaratorPath) {
+  const rval = declaratorPath.get("init")
+                              .evaluate();
+  if (rval.confident === true && rval.value === undefined) {
+    declaratorPath.node.init = null;
+  }
+}
+
+function areAllBindingsNotSeen(declaratorPath, seenNames, t) {
+  const id = declaratorPath.node.id;
+  const names = t.getBindingIdentifiers(id);
+  for (const name in names) {
+    if (seenNames.has(name)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+module.exports = function({ types: t }) {
+  let names = null;
+  let functionNesting = 0;
+  return {
+    name: "remove-undefined-if-possible",
+    visitor: {
+      Program: {
+        enter() {
+          names = new Set();
+          functionNesting = 0;
+        },
+        exit() {
+          names = null;
+          functionNesting = 0;
+        },
+      },
+      Function: {
+        enter() {
+          functionNesting++;
+        },
+        exit() {
+          functionNesting--;
```

Comment:
I suppose you're using this only to check global ? you can use this information to check if different nested function scopes have same variable name and then solve - 

```js
function foo() {
  var x = void 0;
  function bar() {
    var x = void 0; // this is not removed now
  }
}
```